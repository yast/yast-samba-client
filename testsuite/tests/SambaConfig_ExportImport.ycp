/**
 * test SambaConfig::Export:
 * should remove modified flags and internal shares from config
 */
{
  import "Testsuite";
  import "SambaConfig";

  map i = $[
    "a" : $[
      "Bee Bee"         : "x",
      "_modified"       : true,
      "_disabled"       : true,
      "_xxx"            : 8,
      "_comment"        : "A"
    ],
    "_internal"         : $[
      "abc"             : "ABC"
    ],
    "removed"           : nil,
    "b"                 : $[
      "no"              : nil,
      "Two Two"         : 22
    ]
  ];

  Testsuite::Dump (i);

  Testsuite::Test(``(SambaConfig::Import (i)), [ $[], $[], $[] ], 0);

  Testsuite::Test (``(SambaConfig::Export ()), [], 0);
}
