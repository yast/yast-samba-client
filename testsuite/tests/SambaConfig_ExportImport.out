Dump	$["_internal":$["abc":"ABC"], "a":$["Bee Bee":"x", "_comment":"A", "_disabled":true, "_modified":true, "_xxx":8], "b":$["Two Two":22, "no":nil], "removed":nil]
Return	nil
Return	[$["comment":"A", "disabled":true, "name":"a", "parameters":$["bee_bee":"x"]], $["name":"b", "parameters":$["two_two":"22"]]]
